### DISCUSSION TITLE
How to select with an extra computed column

### DISCUSSION THREAD

**Discussion by GrimLothar** [supabase/supabase#11496] [Questions] [answered]
I found[ this discussion](https://github.com/supabase/supabase/discussions/7875), where it is using a computed column to order its results.

But what I'm trying to do is to actually get that computed value as separate column in the row.

Basically, i'm trying to count how many participants are enrolled in a tournament, where participants is a different table.

`await supabase.from('tournaments').select('*,  get_tournament_participant_count')`

Running this gets me this error though: 

```
{
  message: "Could not find a relationship between 'tournaments' and 'get_tournament_participant_count' in the schema cache",
  hint: "Verify that 'tournaments' and 'get_tournament_participant_count' exist in the schema 'public' and that there is a foreign key relationship between them. If a new relationship was created, try reloading the schema cache."
}
```

This is my function:

```
CREATE OR REPLACE FUNCTION public.get_tournament_participant_count(tournamentid integer) RETURNS SETOF BIGINT AS $$
    BEGIN
        RETURN QUERY
            SELECT COUNT(*) FROM public."participants" WHERE "participants".tournament_id = tournamentid;
    END;
$$ LANGUAGE plpgsql;
```

Running that function returns the correct number of participants.

**Comment by silentworks** [comment, marked as answer]
Hi, so looking at this it would seem your function isn't constructed properly, you are passing a `tournamentid` as a property which isn't something you do when using a computed column. You should be passing the table name that you want the computed column on as a property.

```sql
CREATE OR REPLACE FUNCTION public.get_tournament_participant_count(tournaments) RETURNS BIGINT AS $$
    SELECT COUNT(*) FROM public."participants" WHERE "participants".tournament_id = $1.id;
$$ STABLE LANGUAGE sql;
```

I've made an assumption in the above code that your `tournaments` table has an `id` column, the `$1` is a placeholder for the `tournaments` table. I've also changed the language to SQL as there wasn't anything in particular that needed the use of PLPGSQL.

**Reply by GrimLothar** [reply]
Ahhh that worked beautifully! Thank you!
